Serenity-Website/src/MoodBoard/MoodBoard.jsx

```diff
@@ -1,12 +1,127 @@
+
+    const handleMoodSelect = (mood) => {
+        // handle mood changes
+        setSelectedMood(prevMood => prevMood === mood ? null : mood);
+    }
+
+    const handleMoodSubmit = async() => {
+        if(!selectedMood) {
+            alert('Please select a mood');
+            return;
+        }
+
+        try {
+            const response = await fetch (url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                credentials : 'include',
+                body: JSON.stringify ({ mood: selectedMood })
+            });
+
+            if (response.status === 400){
+                setErrorMessage("Mood already logged for today");
+            } else if (!response.ok) {
+                setErrorMessage("Error logging mood");
+                throw new Error('Failed to log mood');
+            } else {
+                const data = await response.json();
+                setMoodSubmitted(true);
+                // set success message if user logs mood successfully
+                setSuccessMessage('You have successfully logged your mood');
+                // clear error message
+                setErrorMessage('');
+                console.log('Mood loged successfully:', data);
+            }
+            
+            } catch(error) {
+            console.error('Error logging mood', error);
+            setErrorMessage("Error logging mood");
+        }
+    };
+
     return (
         <>
-        <p>Hi there</p>
+        <Header />
+        <div className='back-button-div'><button className='back-button' onClick={() => navigate('/activites-feed')}> <FontAwesomeIcon icon={faX} /> </button></div>
+        <div className='mood-entry'>
+            <h2 className="prompt">How are you feeling today?</h2>
+            <h4 className="current-date"><FontAwesomeIcon icon={faCalendar} />   {currentDate}</h4>
+            <div className="mood-options">
+                {Object.keys(moods).map((mood) => (
+                    <span
+                        key={mood}
+                        className={`mood-button ${selectedMood === mood ? 'selected' : ''}`}
+                        onClick={() => handleMoodSelect(mood)}
+                        disabled={moodSubmitted}
+                    >
+                        {moods[mood]} <br /> <span>{mood}</span>
+                    </span>
+                ))}
+            </div>
+            <button onClick={handleMoodSubmit} disabled={moodSubmitted} className='submit-button'>Log Mood</button>
+            
+            {errorMessage && <p className="error-message">{errorMessage}</p>}
+            {successMessage && <p className="success-message">{successMessage}</p>}
+            
+            {moodSubmitted && <button onClick={() => navigate('/past-moods')} className='view-past-moods-button'>View Your Past Moods</button>}
+
+        </div>
         </>
-    )
+    );
 
-}
+};
 
 export default MoodBoard;
```

Comment:
Also, MoodBoard doesn't really make sense as a class name anymore. Let's change it to something like "MoodAndActivityNewEntryPage".